crates/feldera-types/src/checkpoint.rs

```diff
@@ -8,7 +8,17 @@ pub struct CheckpointStatus {
     pub success: Option<u64>,
 
     /// Most recently failed checkpoint, and the associated error.
-    pub failure: Option<(u64, String)>,
+    pub failure: Option<CheckpointFailure>,
+}
+
+/// Information about a failed checkpoint.
+#[derive(Clone, Debug, Default, Serialize, Deserialize, ToSchema)]
```

Comment:
I did this change because the rust rest api generator can;t deal with Option<Tuple> (I think) it generated an empty structure...

Reply:
Explicitly named types are a better interface anyway :+1: